app/src/main/java/com/mdg/droiders/samagra/shush/adapters/TimeListAdapter.java

```diff
@@ -0,0 +1,495 @@
+        String endTimeString = holder.endTime.getText().toString();
+        values.put(PlacesContract.TimeEntry.COLUMN_START_TIME, startTimeString);
+        values.put(PlacesContract.TimeEntry.COLUMN_END_TIME, endTimeString);
+
+        mContext.getContentResolver().update(
+                Uri.withAppendedPath(PlacesContract.TimeEntry.CONTENT_URI, String.valueOf(holder.id))
+                , values, null, null
+        );
+
+        Calendar startTime = Calendar.getInstance();
+        Calendar endTime = Calendar.getInstance();
+        setTimesFromHolder(holder, startTime, endTime);
+        closeCursor();
+        refreshCursor();
+        alarmScheduler.setWeeklyAlarm(startTime, endTime, getDayArr(holder), holder.id);
+    }
+
+    /**
+     * Updates db and sets alarm for the corresponding day.
+     *
+     * @param holder The holder instance that is currently bound to the row whose
+     *               alarm is set.
+     * @param day    The day for which alarm is set. The day is zero indexed and starts from monday.
+     */
+    private void notifyDayAlarmSet(TimeRowHolder holder, int day) {
+
+        ContentValues values = new ContentValues();
+        values.put(getColumnFromDay(day), true);
+        mContext.getContentResolver().update(
+                Uri.withAppendedPath(PlacesContract.TimeEntry.CONTENT_URI, String.valueOf(holder.id))
+                , values, null, null
+        );
+
+        Calendar startTime = Calendar.getInstance();
+        Calendar endTime = Calendar.getInstance();
+        setTimesFromHolder(holder, startTime, endTime);
+        alarmScheduler.setSingleDayAlarm(startTime, endTime, day, holder.id);
+    }
+
+    /**
+     * Updates db and cancels alarm for the corresponding day.
+     *
+     * @param holder The holder instance that is currently bound to the row whose
+     *               alarm is cancelled.
+     * @param day    The day for which alarm is cancelled. The day is zero indexed and starts from monday.
+     */
+    private void notifyDayAlarmCancelled(TimeRowHolder holder, int day) {
+
+        ContentValues values = new ContentValues();
+        values.put(getColumnFromDay(day), false);
+        mContext.getContentResolver().update(
+                Uri.withAppendedPath(PlacesContract.TimeEntry.CONTENT_URI, String.valueOf(holder.id))
+                , values, null, null
+        );
+
+        Calendar startTime = Calendar.getInstance();
+        Calendar endTime = Calendar.getInstance();
+        setTimesFromHolder(holder, startTime, endTime);
+        alarmScheduler.cancelSingleAlarm(day, holder.id);
+    }
+
+    /**
+     * Returns the corresponding column of the day in db.
+     *
+     * @param day The day for which the column is to be returned
+     * @return The column corresponding to the given day
+     */
+    private String getColumnFromDay(int day) {
+        switch (day) {
+            case 0: {
+                return PlacesContract.TimeEntry.COLUMN_MONDAY;
```

Comment:
Why isn't an array of `TimeEntry.<something>` is used and then accessed by its index? That would have been an easier way for this.

Reply:
To remember the names of the columns and to make the code readable this is used. For instance, COLUMN_WEDNESDAY seems more readable than column[2].

Reply:
Are you sure? Currently, your code is spanning over 450 lines and everything is repeated 6 times. Using arrays and enums help you write code faster, read it faster and make it shorter that is what they are for.